Chp9Lsn11/Chp9Lsn11/Program.cs

```diff
@@ -0,0 +1,137 @@
+            string userCommand = TextMenu(answer);
+            MultipleType FinalAnswer = CallOption(userCommand);
+            PrintAnswer(userCommand, FinalAnswer);
+            Console.ReadLine();
+        }
+
+        public static string TextMenu(string answer)
+        {
+            bool resolved = false;
+            
+            Console.WriteLine("This program has three options. Option 'reverse' allows you to enter a series of numbers ");
+            Console.WriteLine("that are returned in reverse order. Option 'average' calculates the average of a series of");
+            Console.WriteLine("numbers. Option 'equation' calculates x of the linear equation. Type the name of the option");
+            Console.WriteLine("you desire and press enter.");
+            Console.WriteLine("");
+
+            if(answer != "Error")
+            {
+                resolved = true;
+            }
+
+            while (resolved == false)
+            {
+                string input = Console.ReadLine();
+
+                if (input != "reverse" && input != "average" && input != "equation")
+                {
+                    Console.WriteLine("");
+                    Console.Write("Invalid Input!");
+                }
+                else
+                {
+                    answer = input;
+                    resolved = true;
+                }
+            }
+
+            return answer;
+        }
+
+        public static MultipleType CallOption(string optionSelect)
+        {
+            MultipleType userInput = new MultipleType();
+
+            if (optionSelect == "reverse")
+            {
+                userInput.integer = UserInput.UserString();
+                userInput.integer = ComputerCalculate(userInput.integer);
+            }
+
+            if (optionSelect == "average")
+            {
+                userInput.numberSequence = UserInput.UserArray();
+                userInput.numberSequence = ComputerCalculate(userInput.numberSequence);
+            }
+
+            if (optionSelect == "equation")
+            {
+                userInput.threeDouble = UserInput.UserDoubles();
+                userInput.x = ComputerCalculate(userInput.threeDouble.a, userInput.threeDouble.b, userInput.threeDouble.c);
+            }
+
+            return userInput;
+        }
+
+        public static string ComputerCalculate(string integer)
+        {
+            char[] integerArray = integer.ToCharArray();
+            Array.Reverse(integerArray);
+            return new string(integerArray);
+        }
+
+        public static double[] ComputerCalculate(double[] numberSequence)
+        {
+            double sum = 0;
+            int numColumns = numberSequence.GetLength(0);
+
+            for(int i = 0; i < numColumns; i++)
+            {
+                sum += numberSequence[i];
+            }
+
+            double average = sum / numColumns;
+            double[] averageArray = { average };
+            return averageArray;
```

Comment:
Why is the answer being returned as an array?  What purpose?  When possible the code should be self-documenting; what are you trying to tell me by returning an array?